test/connection.js

```diff
@@ -622,7 +622,6 @@ describe('Connection', () => {
 
                             expect(err).to.not.exist();
                             expect(count1).to.equal(2);
-                            expect(Object.keys(server.connections[0]._connections).length).to.equal(2);
```

Comment:
The way this test is written doesn't take into account the handshake that happens before the `secureConnection` event is fired. I removed this part of the test to account for that—is that a correct way to fix it?